Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -274,67 +300,175 @@ def TestConfigUpdate(Contents):
         retval = -1
     else:
         retval = WriteFile(destFileFullPath, Contents)
-        if retval == 0 and os.path.exists(AGENT_RESOURCE_VERSION_PATH): #notify server only if plugin is present
-            LG().Log(LogType.Info, 'Updated the file, going to notify server')
+        LOG_ACTION.log(LogType.Info, 'Updated configuration file for ' + ConfigID)
+        # Notify NPM of config change if single workspace scenario
+        if (not retval) and len(GetWorkspaceListFromStateFile()) == 1:
             retval = NotifyServer(Commands.Config)
     return retval
 
+# Needs to run after state file updation in set
 def SetFilesUpdate(newVersion):
+    wsRunningNPM = GetWorkspaceRunningNPMAgent()
     retval = UpdateAgentBinary(newVersion)
     retval &= UpdatePluginFiles()
+    if wsRunningNPM is not None:
+        retval &= UpdatePluginConfFiles(wsRunningNPM)
     if retval:
         return 0
     return -1
 
+def UpdateStateFile(completeList, removeList, owner):
+    retval = 0
+    if len(completeList) != 0 and (owner is not None) and (owner in completeList):
+        listToWrite = []
+        listToWrite.append(owner)
+        for x in completeList:
+            if x != owner and x not in removeList:
+                listToWrite.append(x)
+        contents = ''
+        for x in listToWrite:
+            contents += x
+            contents += '\n'
+        retval = WriteFile(os.path.join(PATH_NPM_STATE, FNAME_NPM_STATE), contents)
+    else:
+        retval = -1
+    return retval
+
 def UpdateAgentBinary(newVersion):
     retval = True
     arch = platform.architecture()
     src = ''
     if arch is not None and arch[0] == X64:
         src = RESOURCE_MODULE_PATH.__add__(DSC_X64_AGENT_PATH)
-        retval &= DeleteAllFiles(src, AGENT_BINARY_PATH)
-        retval &= CopyAllFiles(src, AGENT_BINARY_PATH)
+        retval &= DeleteAllFiles(src, PATH_AGENT_BINARY)
+        retval &= CopyAllFiles(src, PATH_AGENT_BINARY)
     else:
         src = RESOURCE_MODULE_PATH.__add__(DSC_X86_AGENT_PATH)
-        retval &= DeleteAllFiles(src, AGENT_BINARY_PATH)
-        retval &= CopyAllFiles(src, AGENT_BINARY_PATH)
+        retval &= DeleteAllFiles(src, PATH_AGENT_BINARY)
+        retval &= CopyAllFiles(src, PATH_AGENT_BINARY)
 
     # set capabilities to binary
     src_files = os.listdir(src)
     for file_name in src_files:
-        full_file_name = os.path.join(AGENT_BINARY_PATH, file_name) # assuming only file in directory
+        full_file_name = os.path.join(PATH_AGENT_BINARY, file_name) # assuming only file in directory
         break
     retval &= NPM_ACTION.binary_setcap(full_file_name)
 
-    # Notify ruby plugin
-    #retval &= NotifyServer(Commands.RestartNPM)
-
     #Update version number
     if retval == True:
-        WriteFile(AGENT_RESOURCE_VERSION_PATH, newVersion)
+        WriteFile(os.path.join(PATH_NPM_STATE, FNAME_NPM_VERSION), newVersion)
```

Comment:
Is this directory guaranteed to exist on all paths where `UpdateAgentBinary`is called?